spec/statements.md

```diff
@@ -566,102 +683,126 @@ Within a switch block, the meaning of a name used in an expression context shall
+- The while statement is reachable and the Boolean expression does not have the constant value true.
 
 ### The do statement
 
 The do statement conditionally executes an embedded statement one or more times.
 
-[]{#Grammar_do_statement .anchor}do-statement:\
-*do* embedded-statement *while* *(* boolean-expression *)* *;*
+[#Grammar_do_statement .anchor](#Grammar_do_statement .anchor)do-statement:\
+
+```ANTLR
+do embedded-statement while ( boolean-expression ) ;
+```
 
 A do statement is executed as follows:
 
--   Control is transferred to the embedded statement.
+- Control is transferred to the embedded statement.
 
--   When and if control reaches the end point of the embedded statement (possibly from execution of a continue statement), the *boolean-expression* (§12.21) is evaluated. If the Boolean expression yields true, control is transferred to the beginning of the do statement. Otherwise, control is transferred to the end point of the do statement.
+- When and if control reaches the end point of the embedded statement (possibly from execution of a continue statement), the *boolean-expression* (§12.21) is evaluated. If the Boolean expression yields true, control is transferred to the beginning of the do statement. Otherwise, control is transferred to the end point of the do statement.
 
 Within the embedded statement of a do statement, a break statement (§13.10.2) may be used to transfer control to the end point of the do statement (thus ending iteration of the embedded statement), and a continue statement (§13.10.3) may be used to transfer control to the end point of the embedded statement (thus performing another iteration of the do statement).
 
 The embedded statement of a do statement is reachable if the do statement is reachable.
 
-[[]{#_Ref470173280 .anchor}]{#_Toc445783028 .anchor}The end point of a do statement is reachable if at least one of the following is true:
+[#_Ref470173280 .anchor](#_Ref470173280 .anchor)[#_Toc445783028 .anchor](#_Toc445783028 .anchor)The end point of a do statement is reachable if at least one of the following is true:
 
--   The do statement contains a reachable break statement that exits the do statement.
+- The do statement contains a reachable break statement that exits the do statement.
 
--   The end point of the embedded statement is reachable and the Boolean expression does not have the constant value true.
+- The end point of the embedded statement is reachable and the Boolean expression does not have the constant value true.
 
 ### The for statement
 
 The for statement evaluates a sequence of initialization expressions and then, while a condition is true, repeatedly executes an embedded statement and evaluates a sequence of iteration expressions.
 
-[]{#Grammar_for_statement .anchor}for-statement:\
-*for* *(* for-initializer~opt~ *;* for-condition~opt~ *;* for-iterator~opt~ *)* embedded-statement
+[#Grammar_for_statement .anchor](#Grammar_for_statement .anchor)for-statement:\
```

Comment:
A single ANTLR block should start on this line and include everything up to line 805.